overlord/snapstate/handlers.go

```diff
@@ -369,6 +378,13 @@ func updatePrereqIfOutdated(t *state.Task, snapName string, contentAttrs []strin
 	ts, err := UpdateWithDeviceContext(st, snapName, nil, userID, flags, deviceCtx, "")
 	if err != nil {
 		if conflErr, ok := err.(*ChangeConflictError); ok {
+			// If we aren't seeded, then it's to early to do any updates and we cannot
+			// handle this during seeding, so expect the ChangeConflictError in this scenario.
+			if seeded, err := isSeeded(st); err != nil || !seeded {
```

Comment:
Instead of doing this check we could also use `DevicePastSeeding` when retrieving the device context in line 372, and check that way instead of checking once we get the error. Do anyone have any opinion on this?

Reply:
I think we could check instead directly that conflErr.ChangeKind == "seed" instead of calling isSeededAgain? otherwise if we want to keep the code as is we shouldn't ignore errors from isSeeded

Reply:
I like the simplicity of just checking ChangeKind, but it just seemed a bit fragile. I updated to this approach!

Reply:
Yes, it's a bit fragile but you have test with all the moving parts, and they are all in this package. If the moving parts were crossing packages I would be more worried about the fragility.